insights/parsers/nftables.py

```diff
@@ -0,0 +1,154 @@
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+    Attributes:
+        main_data (dict): A dict contains the address_family/table/chain/rule data.
+
+    Raises:
+        SkipComponent: No available nftables.
+
+    Examples:
+        >>> type(nft_obj)
+        <class 'insights.parsers.nftables.NftListRuleSet'>
+        >>> [str(key) for key in nft_obj.main_data.keys()]  # to be compatible with python2.6/2.7, transform unicode to string
+        ['ip']
+        >>> [str(key) for key in nft_obj.main_data['ip'].keys()] # to be compatible with python2.6/2.7, transform unicode to string
+        ['example_table1']
+        >>> 'example_chain1' in nft_obj.chains('ip', 'example_table1')
+        True
+        >>> len(nft_obj.rules('ip', 'example_table1', 'example_chain1'))
+        2
+
+    """
+
+    def parse_content(self, content):
+        """
+        There are a lot of other types in nftables besides "chain/rule", like "map", "set", "quota".
+        But they are not extracted now since there are no rules dependent on them. If needed,
+        we can extract them later.
+        """
+        super(NftListRuleSet, self).parse_content(content)
+        self.main_data = defaultdict(dict)
+        for item in self.data['nftables']:
+            for key, value in item.items():
+                address_name = value.get('family')
+                if key == 'table':
+                    table_name = value['name']
+                    self.main_data[address_name][table_name] = defaultdict(dict)
+                else:
+                    if key in ['chain', 'rule']:
+                        table_name = value['table']
+                        if key == 'chain':
+                            chain_name = value['name']
+                            self.main_data[address_name][table_name]['chains'][chain_name] = {}
+                            self.main_data[address_name][table_name]['chains'][chain_name]['attrs'] = value
+                            self.main_data[address_name][table_name]['chains'][chain_name]['rules'] = []
+                        else:
+                            # rule
+                            chain_name = value['chain']
+                            self.main_data[address_name][table_name]['chains'][chain_name]['rules'].append(value)
+
+        if not self.main_data:
+            raise SkipComponent
+
+    def tables(self, address_family):
+        """dict: Return the tables in some address family."""
+        return self.main_data[address_family]
+
+    def chains(self, address_family, table_name):
+        """dict: Returns the chains in some table."""
+        return self.main_data[address_family][table_name]['chains']
+
+    def rules(self, address_family, table_name, chain_name):
+        """list: Returns the rules of some chain in some table."""
+        return self.main_data[address_family][table_name]['chains'][chain_name]['rules']
```

Comment:
With the datatype `defaultdict(dict)` for this `self.main_data` in some inside levels, the directly data access could keeps adding new elements into this `self.main_data` in the later parser usage, eg: 
```
>>> main_data = defaultdict(dict)
>>> main_data
defaultdict(<class 'dict'>, {})
>>> tables = main_data["some_unexist_address_family"]
>>> tables
{}
>>> main_data
defaultdict(<class 'dict'>, {'some_unexist_address_family': {}})
```
@huali027  Shall we consider to convert the datatype of `self.main_data` back to `dict` at the end(Line#140 ) of `parse_content()` ? Could be something like: 

```
        for addr, table in self.main_data.items():
            for k, v in table.items():
                table[k] = dict(v)
        self.main_data = dict(self.main_data)
``` 

And if this is the case, shall we also add some test cases for the `unexist` key access case?